packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Relations/RelationModal.tsx

```diff
@@ -154,6 +157,17 @@ function reducer(state: State, action: Action): State {
         ...state,
         hasUnsavedChanges: action.payload.hasUnsavedChanges,
       };
+    case 'REMOVE_CREATE_FROM_HISTORY':
+      const lastIndex = state.documentHistory.length - 1;
+      const lastDocument = state.documentHistory[lastIndex];
+      const isLastDocumentCreate =
+        !lastDocument.documentId && lastDocument.collectionType !== SINGLE_TYPES;
+      return {
+        ...state,
+        documentHistory: isLastDocumentCreate
+          ? state.documentHistory.slice(0, lastIndex)
+          : state.documentHistory,
+      };
```

Comment:
I think you can rely on GO_BACK instead for this instead

Reply:
yeah but the GO BACK change also the document, in my case I just need to clean up the history and replace the object with documentId undefined (basically the not saved new relation) with the object containing the documentID (and model and collection type) coming from the create/publish API response

Reply:
What I mean is that conceptually, the intent is still to go back, and it's triggered on the back button. With reducers we to think about events first, that's why I think it would be nice for it to be the same event.

The logic in terms of how to react and adapt state from this event can be figured out in the GO_BACK reducer implementation. You have access to the previous state, so you can infer if the user was creating when he clicked back, so you can add an if clause based on that